docs/guides/data/mounting_data.md

```diff
@@ -1,105 +1,192 @@
+you want your code to be able to access a dataset during the job, you need to
+mount the dataset to the server where the job is running (just like you need to
+plug a flash drive into your computer in order to access its files). Mounting
+datasets to a job is easy: just pass the `--data` flag to the `floyd run` cli
+command as detailed below.
+
+### The `--data` flag
+
+To properly use the `--data` flag with `floyd run`, you need to specify two
+things:
+
+  1. The name of the dataset you want to mount.
+  2. A name for the folder where the data will be
+  accessible to your code during the job, we call this the "mount point". You
+  can give this folder (mount point) any name you want. Don't precede this name
+  with a forward slash `/`.
+
+These two things are separated by a `:` with no spaces. Here's the syntax:
 
-## Datasets
-
-Floyd datasets are directories of data files that can be used in a project run. To create a new dataset, please follow 
-[this guide](../basics/create_new/#create-a-new-dataset). You can view the datasets you 
-have created in the [datasets page](https://www.floydhub.com/datasets) in the dashboard. You can also view 
-public datasets by searching for it on FloydHub.
-
-## Mounting datasets
+```
+floyd run --data <data_set_name>:<mount_point> …(rest of run command)
+```
 
-### Overview
+Let's go through a couple of examples to show how to mount one or more datasets to your job.
 
-You can mount one or more datasets when you run a job using the [floyd run](../../commands/run) command. 
-This automatically makes the mounted datasets available for you to use in your job, without having to download them each time.
+#### Example 1
 
-To mount a specific version of a dataset, you need to specify its [full name](#datasource-name) and the [mount point](#mount-point). 
-The syntax is `--data <data_name>:<mount_point>`
+The command below will mount FloydHub's public
+[Udacity GAN](https://www.floydhub.com/floydhub/datasets/udacity-gan)
+ dataset at `/my_data`
+```
+floyd run --data floydhub/datasets/udacity-gan:my_data "python my_script.py"
+```
 
-For example, to mount the [VGG 19-layers](https://www.floydhub.com/floydhub/datasets/vgg-ilsvrc-19-layers/1) dataset under `/vgg`:
-```bash
+A couple of things to note:
+
+  - There is no space between the name of the dataset
+    (`floydhub/datasets/udacity-gan`) and the mount point name (`my_data`).
+  - A colon (`:`) is used to separate the name of the dataset and the
+    mount point.
+  - The mount point name does not have have a preceding `/`, but your code
+    needs to add the preceding slash to locate the dataset during the job.
+    This means that if you specify `my_data` as the mount point, your code can
+    find your dataset at `/my_data`. If you specify `mnist_data` as the mount
+    point, your code needs to look in `/mnist_data` for your data.
+  - Nested mount points are not supported. This means a mount point like
+    `my_data/foo` will not work.
+
+!!! important
```

Comment:
@houqp Is this still the case? Can we just not strip any preceeding forward slashes to simplify it for the user?

Reply:
We always mount the dataset at the root directory. We can simplify it by also symlinking that mounted directory to current working directory, but that might cause directory naming conflict if the uploaded code directory also happened to have a directory with the same name. I think it's a a reasonable tradeoff to just fail the job if that's the case.

Reply:
Yeah I guess this has been fixed. I'll update this before merging

Reply:
No, that's not what I meant. Currently the mount_point, cannot have a preceding slash. E.g. `abc` is a valid mount point, but `/abc` is not (throws an Internal Sever Error). I think we should allow both, since the final mount point is indeed `/abc`

Reply:
oh, that's already fixed in the backend. i striped all preceeding slashes.

Reply:
My reply was to @houqp. Awesome if this has already been fixed!!